### ISSUE TITLE
Separate holdings records generate the same UUID

### ISSUE THREAD

**Issue by jermnelson** [FOLIO-FSE/folio_migration_tools#397]
In a recent migration, one of our analysts discovered an error involving the correct Holdings being mapped to its items. We are using the `HoldingsCsvTransformer` transformer (with HRID handling set to default using the default `holdings_merge_criteria` of "instanceId", "permanentLocationid", and "callNumber").

From our incoming tsv_file (the **CATKEY** is the legacy_code we match from the MARC 21 bib file):

<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CATKEY</th>
      <th>FORMAT</th>
      <th>CALL_NUMBER_TYPE</th>
      <th>BASE_CALL_NUMBER</th>
      <th>VOLUME_INFO</th>
      <th>BARCODE</th>
      <th>LIBRARY</th>
      <th>HOMELOCATION</th>
      <th>CURRENTLOCATION</th>
      <th>ITEM_TYPE</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>10435</th>
      <td>a660592</td>
      <td>SERIAL</td>
      <td>LC</td>
      <td>B3212 .Z7 A12</td>
      <td>V.12 2006/2009</td>
      <td>36105132152419</td>
      <td>GREEN</td>
      <td>STACKS</td>
      <td>STACKS</td>
      <td>STKS-MONO</td>
    </tr>
    <tr>
      <th>10436</th>
      <td>a660592</td>
      <td>SERIAL</td>
      <td>LCPER</td>
      <td>B3212 .Z7 A12</td>
      <td>V.10-11 2002/03-2004/05</td>
      <td>36105123807179</td>
      <td>GREEN</td>
      <td>STACKS</td>
      <td>STACKS</td>
      <td>STKS-MONO</td>
    </tr>
    <tr>
      <th>10437</th>
      <td>a660592</td>
      <td>SERIAL</td>
      <td>LCPER</td>
      <td>B3212 .Z7 A12</td>
      <td>V.1-2 1988-1989</td>
      <td>36105008356383</td>
      <td>GREEN</td>
      <td>STACKS</td>
      <td>STACKS</td>
      <td>STKS-MONO</td>
    </tr>
    <tr>
      <th>10438</th>
      <td>a660592</td>
      <td>SERIAL</td>
      <td>LCPER</td>
      <td>B3212 .Z7 A12</td>
      <td>V.13 2015</td>
      <td>36105222580610</td>
      <td>GREEN</td>
      <td>STACKS</td>
      <td>STACKS</td>
      <td>STKS-MONO</td>
    </tr>
    <tr>
      <th>10439</th>
      <td>a660592</td>
      <td>SERIAL</td>
      <td>LCPER</td>
      <td>B3212 .Z7 A12</td>
      <td>V.14 2016</td>
      <td>36105226756356</td>
      <td>GREEN</td>
      <td>STACKS</td>
      <td>STACKS</td>
      <td>STKS-MONO</td>
    </tr>
    <tr>
      <th>10440</th>
      <td>a660592</td>
      <td>SERIAL</td>
      <td>LCPER</td>
      <td>B3212 .Z7 A12</td>
      <td>V.15:PT.1 2017</td>
      <td>36105227374167</td>
      <td>GREEN</td>
      <td>STACKS</td>
      <td>STACKS</td>
      <td>STKS-MONO</td>
    </tr>
    <tr>
      <th>10441</th>
      <td>a660592</td>
      <td>SERIAL</td>
      <td>LCPER</td>
      <td>B3212 .Z7 A12</td>
      <td>V.16 2018</td>
      <td>36105228622564</td>
      <td>GREEN</td>
      <td>STACKS</td>
      <td>STACKS</td>
      <td>STKS-MONO</td>
    </tr>
    <tr>
      <th>10442</th>
      <td>a660592</td>
      <td>SERIAL</td>
      <td>LCPER</td>
      <td>B3212 .Z7 A12</td>
      <td>V.17 2019/2020</td>
      <td>36105234057490</td>
      <td>GREEN</td>
      <td>STACKS</td>
      <td>STACKS</td>
      <td>STKS-MONO</td>
    </tr>
    <tr>
      <th>10443</th>
      <td>a660592</td>
      <td>SERIAL</td>
      <td>LCPER</td>
      <td>B3212 .Z7 A12</td>
      <td>V.3-4 1990-1993</td>
      <td>36105016681236</td>
      <td>GREEN</td>
      <td>STACKS</td>
      <td>STACKS</td>
      <td>STKS-MONO</td>
    </tr>
    <tr>
      <th>10444</th>
      <td>a660592</td>
      <td>SERIAL</td>
      <td>LCPER</td>
      <td>B3212 .Z7 A12</td>
      <td>V.5-6 1994-1996</td>
      <td>36105020941410</td>
      <td>GREEN</td>
      <td>STACKS</td>
      <td>STACKS</td>
      <td>STKS-MONO</td>
    </tr>
    <tr>
      <th>10445</th>
      <td>a660592</td>
      <td>SERIAL</td>
      <td>LCPER</td>
      <td>B3212 .Z7 A12</td>
      <td>V.7 1997</td>
      <td>36105021595322</td>
      <td>GREEN</td>
      <td>STACKS</td>
      <td>STACKS</td>
      <td>STKS-MONO</td>
    </tr>
    <tr>
      <th>10446</th>
      <td>a660592</td>
      <td>SERIAL</td>
      <td>LCPER</td>
      <td>B3212 .Z7 A12</td>
      <td>V.8-9 1998/1999-2000/2001</td>
      <td>36105111533670</td>
      <td>GREEN</td>
      <td>STACKS</td>
      <td>STACKS</td>
      <td>STKS-MONO</td>
    </tr>
    <tr>
      <th>10447</th>
      <td>a660592</td>
      <td>SERIAL</td>
      <td>LC</td>
      <td>XX(660592.13)</td>
      <td>V.17</td>
      <td>36105230133121</td>
      <td>GREEN</td>
      <td>STACKS</td>
      <td>ON-ORDER</td>
      <td>STKS-MONO</td>
    </tr>
  </tbody>
</table>

The `HoldingsCsvTransformer` produces the following two Holdings records that we would expect with the default merge criteria:

```javascript
{'id': '81d1c667-06c0-528c-ab7b-36685d09f7ab',
 'metadata': {'createdDate': '2022-10-17T21:55:36.683',
  'createdByUserId': '8cc3ab86-c943-4d53-8df7-b3dc64fb44ee',
  'updatedDate': '2022-10-17T21:55:36.683',
  'updatedByUserId': '8cc3ab86-c943-4d53-8df7-b3dc64fb44ee'},
 'formerIds': ['a660592'],
 'instanceId': '3191fabe-e494-5757-b9d1-3b98efa4cfab',
 'permanentLocationId': '4573e824-9273-4f13-972f-cff7bf504217',
 'callNumberTypeId': '95467209-6d7b-468b-94df-0f5d7ad2747d',
 'callNumber': 'B3212 .Z7 A12',
 'holdingsTypeId': '03c9c400-b9e3-4a07-ac0e-05ab470233ed',
 'sourceId': 'f32d531e-df79-46b3-8932-cdd35f7a2264',
 'notes': []}
```

```javascript
{'id': '81d1c667-06c0-528c-ab7b-36685d09f7ab',
 'metadata': {'createdDate': '2022-10-17T21:55:36.685',
  'createdByUserId': '8cc3ab86-c943-4d53-8df7-b3dc64fb44ee',
  'updatedDate': '2022-10-17T21:55:36.685',
  'updatedByUserId': '8cc3ab86-c943-4d53-8df7-b3dc64fb44ee'},
 'formerIds': ['a660592'],
 'instanceId': '3191fabe-e494-5757-b9d1-3b98efa4cfab',
 'permanentLocationId': '4573e824-9273-4f13-972f-cff7bf504217',
 'callNumberTypeId': '95467209-6d7b-468b-94df-0f5d7ad2747d',
 'callNumber': 'XX(660592.13)',
 'holdingsTypeId': '03c9c400-b9e3-4a07-ac0e-05ab470233ed',
 'sourceId': 'f32d531e-df79-46b3-8932-cdd35f7a2264'}
```

## However the UUID for both records are the same!
This is causing an issue later with items like the following:

```javascript
{'id': 'c74453f8-bd80-58a8-a992-779411ebee1a',
 'metadata': {'createdDate': '2022-10-17T21:55:51.582',
  'createdByUserId': '8cc3ab86-c943-4d53-8df7-b3dc64fb44ee',
  'updatedDate': '2022-10-17T21:55:51.582',
  'updatedByUserId': '8cc3ab86-c943-4d53-8df7-b3dc64fb44ee'},
 'holdingsRecordId': '81d1c667-06c0-528c-ab7b-36685d09f7ab',
 'barcode': '36105021595322',
 'enumeration': 'V.7 1997',
 'status': {'name': 'Available', 'date': '2022-10-17T21:55:51.582422'},
 'materialTypeId': '1a54b431-2e4f-452d-9cae-9cee66c9a892',
 'permanentLoanTypeId': '2b94c631-fca9-4892-a730-03ee529ffe27',
 'permanentLocationId': '4573e824-9273-4f13-972f-cff7bf504217',
 '_version': 1}
```
should be matched to first holding record but not the second.

**Comment by bltravis** [comment]
@jermnelson Have you tried using the field concatenation approach for this?

eg (in holdings field map):
`{
        "folio_field": "legacyIdentifier",
        "legacy_field": "CATKEY",
        …
},
{
        "folio_field": "legacyIdentifier",
        "legacy_field": "BASE_CALL_NUMBER",
        …
}`

(in item field map):
`{
        "folio_field": "holdingsId",
        "legacy_field": "CATKEY",
        …
},
{
        "folio_field": "holdingsId",
        "legacy_field": "BASE_CALL_NUMBER",
        …
}`

This **should** concatenate the contents of those fields (with a " ") into the legacyIdentifer/holdingsId values, respectively.

**Comment by jermnelson** [comment]
@bltravis, thanks for the suggestion! We eventually ended up going with a similar approach in our holdings mapping file by concatenating the `CATKEY`, 'CALL_SEQ`, and `COPY` columns.  

We did find in order for the item field map to work properly we had to also map these fields to the formerId[0] in the holdings field map for the holding_id_map.json file to correctly map these fields for the Item transformer due to this implicated code:
https://github.com/FOLIO-FSE/folio_migration_tools/blob/6ae0a3d1518d6f810c74bfa101939d289d5efea0/src/folio_migration_tools/migration_tasks/holdings_csv_transformer.py#L234-L241

Maybe this should be `"legacyIdentifier` instead of `formerIds` on line 235?
